libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFJsonUtils.h

```diff
@@ -67,7 +67,16 @@
 
 
 /**
- Pull a value from the json-derived object by path ("." delimited)
+ * Pull a value from the json-derived object by path ("." delimited)
+ *
+ * Examples (in pseudo code):
+ *
+ * json = {"a": {"b": [{"c":"xx"}, {"c":"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]}] }}
+ * projectIntoJson(jsonObj, "a") = {"b": [{"c":"xx"}, {"c":"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]} ]}
+ * projectIntoJson(json, "a/b") = [{c:"xx"}, {c:"xy"}, {"d": [{"e":1}, {"e":2}]}, {"d": [{"e":3}, {"e":4}]}]
+ * projectIntoJson(json, "a/b/c") = ["xx", "xy"]                                     // new in 4.1
```

Comment:
I'm wondering about the case, in this example, of:

```
{"a": {"b": [{"c":"xx"}, {"c":"xy"}, [{"c":"xz"}, {"d":"yx"}]]}}
```

Is there any way to target the inner "c" element in the array?  It feels like overloading the delimiter with the same behavior for dictionaries *and* arrays could be problematic.